commons/Common.ml

```diff
@@ -1231,6 +1260,13 @@ let sort_by_key_highfirst xs =
 let sort_by_key_lowfirst xs =
   sort_prof (fun (k1,_v1) (k2,_v2) -> compare k1 k2) xs
 
+let flatten xss =
```

Comment:
also, add tail-rec comment.
Actually I think ocaml now has an attribute like [@@tail-rec] that is checked by the compiler.

Reply:
There is `[@tailcall]` but it's a bit cumbersome because you need to annotate every recursive call.